src/main/java/space/obminyashka/items_exchange/controller/UserController.java

```diff
@@ -181,13 +182,17 @@ public List<ChildDto> updateChildren(@Size(max = MAX_CHILDREN_AMOUNT, message =
     @ApiOperation(value = "Set a new user's avatar image")
     @ApiResponses(value = {
             @ApiResponse(code = 202, message = "ACCEPTED"),
+            @ApiResponse(code = 400, message = "BAD REQUEST"),
             @ApiResponse(code = 403, message = "FORBIDDEN"),
             @ApiResponse(code = 406, message = "NOT ACCEPTABLE"),
             @ApiResponse(code = 415, message = "UNSUPPORTED MEDIA TYPE")})
     @ResponseStatus(HttpStatus.ACCEPTED)
-    public Map<String, byte[]> updateUserAvatar(@RequestParam MultipartFile image, @ApiIgnore Authentication authentication) {
+    public Map<String, byte[]> updateUserAvatar(@RequestParam(name = "image") MultipartFile[] images, @ApiIgnore Authentication authentication) throws BadRequestException {
+        if(images.length != 1){
+            throw new BadRequestException("Image should be only one");
+        }
         User user = getUser(authentication.getName());
-        byte[] newAvatarImage = imageService.compress(image);
+        byte[] newAvatarImage = imageService.compress(images[0]);
```

Comment:
It looks dangerous

Reply:
If length does not equal 1, the endpoint will return a 400 error, so line 195 always has length which equals 1.